### PR TITLE
fix(TDI-39124): first part of fix for new data flow

### PR DESCRIPTION
**Pull request by mbasiuk-talend**
**What is the current behavior?** (You can also link to an open issue here)
Broken converting BigDecimal and Date types
https://jira.talendforge.org/browse/TDI-39124

**What is the new behavior?**
Fixed broken converters

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/779/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/779/)

File: components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceWriterTestIT.java [hunk 1]
```diff
@@ -76,6 +83,13 @@
 
     SalesforceRuntimeTestUtil runtimeTestUtil = new SalesforceRuntimeTestUtil();
 
+    static Schema getFormattedDateField(String name) {
+        Schema schema = AvroUtils._long();
+        LogicalTypes.timestampMillis().addToSchema(schema);
+        Schema.Field field = new Field(name, schema, null, (String) null);
+        return field.schema();
```
Comment (gonchar-ivan) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceWriterTestIT.java]: 1. It is recommended to use AvroUtils._logicalTimestamp()

2. I think it is redundant to create field and then retrieve schema from it. You may just return schema

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceAvroRegistry.java [hunk 2]
```diff
@@ -226,7 +226,7 @@ private Schema inferSchemaField(Field field) {
             return new StringToDoubleConverter(f);
         } else if (AvroUtils.isSameType(fieldSchema, AvroUtils._int())) {
             return new StringToIntegerConverter(f);
-        } else if (AvroUtils.isSameType(fieldSchema, AvroUtils._date())) {
+        } else if (AvroUtils.isSameType(fieldSchema, AvroUtils._date()) || LogicalTypes.timestampMillis() == fieldSchema.getLogicalType()) {
```
Comment (gonchar-ivan) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceAvroRegistry.java]: It is recommended to use LogicalTypeUtils.isLogicalTimestampMillis()

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java [hunk 3]
```diff
@@ -257,47 +258,36 @@ private SObject createSObjectForUpsert(IndexedRecord input) {
...
-            valueToAdd = value;
-            break;
-        }
-        if (valueToAdd instanceof Date) {
-            xmlObject.setField(fieldName, SalesforceRuntime.convertDateToCalendar((Date) valueToAdd));
+        fieldSchema = AvroUtils.unwrapIfNullable(fieldSchema);
+        String javaClassFlag = fieldSchema.getProp(SchemaConstants.JAVA_CLASS_FLAG);
+        if (LogicalTypes.timestampMillis() == fieldSchema.getLogicalType()
+                || "java.util.Date".equals(javaClassFlag)) {
+            // Add Pattern handling
```
Comment (gonchar-ivan) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: Why pattern handling is required here?

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java [hunk 4]
```diff
@@ -257,47 +258,36 @@ private SObject createSObjectForUpsert(IndexedRecord input) {
         return so;
     }
 
-    private void addSObjectField(XmlObject xmlObject, Schema.Type expected, String fieldName, Object value) {
-        Object valueToAdd = null;
+    private void addSObjectField(XmlObject xmlObject, Schema fieldSchema, String fieldName, Object value) {
```
Comment (gonchar-ivan) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: This method requires further refactoring

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/781/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/781/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/782/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/782/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/783/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/783/)

**Pull request merged**